gnd/src/main/java/com/google/android/gnd/ui/common/ResponseValidator.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.ui.common;
+
+import android.content.res.Resources;
+import com.google.android.gnd.GndApplication;
+import com.google.android.gnd.R;
+import com.google.android.gnd.model.form.Field;
+import com.google.android.gnd.model.observation.Response;
+import java8.util.Optional;
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+@Singleton
+public class ResponseValidator {
+
+  private final Resources resources;
+
+  @Inject
+  ResponseValidator(GndApplication application) {
+    this.resources = application.getResources();
+  }
+
+  // TODO: Check valid response values
+  @Nullable
+  public String validate(Field field, Optional<Response> response) {
```

Comment:
We've been using `Optional` instead of `null` for internal APIs. Can we do the same here for consistency?

Reply:
Thanks! Done 2c82209